cmd/atlas/internal/lint/run.go

```diff
@@ -86,7 +86,11 @@ func (r *Runner) summary(ctx context.Context) error {
 		r.sum.Files = append(r.sum.Files, &FileReport{Name: migrate.HashFileName, Error: err.Error()})
 		return r.sum.StepError(stepIntegrityCheck, fmt.Sprintf("File %s is invalid", migrate.HashFileName), err)
 	default:
-		r.sum.StepResult(stepIntegrityCheck, fmt.Sprintf("File %s is valid", migrate.HashFileName), nil)
+		// If the hash file exists, it is valid.
+		if f, err := r.Dir.Open(migrate.HashFileName); err == nil {
+			f.Close()
+			r.sum.StepResult(stepIntegrityCheck, fmt.Sprintf("File %s is valid", migrate.HashFileName), nil)
+		}
```

Comment:
Why this code is needed if we handle ErrChecksumNotFound?

Reply:
Because the dir is empty, the original logic of Validate was allow an empty dir is valid.

This test case run on an empty dir. So it report atlas.sum is valid, but the file is not existing.